evennia/contrib/aisystem/README.md

```diff
@@ -0,0 +1,661 @@
+                      completion, the parallel node's return status being the
+                      same as that of the primary child.
+
+req\_successes (int) - used by parallel nodes; represents the number of child
+                      nodes that must return success for the parallel node
+                      itself to immediately return success.
+
+req\_failures (int) - used by parallel nodes; represents the number of child
+                     nodes that must return success for the parallel node
+                     itself to immediately return success.
+
+default\_success (bool) - used by parallel nodes to decide whether to return
+                         success when all child nodes have completed but
+                         no other condition has been met for returning a status
+
+running (bool) - whether or not the node is currently running or was running
+                 during its last tick.
+
+running\_child (int) - the currently running child of a MemSequence or
+                      MemSelector node.
+
+ticks (int) - for repeater nodes, this is the number of repeats that have
+              already been performed since the repeater started running. For
+              limiter nodes, this is the number of repeats left before the
+              limiter shuts down.
+
+watchers (list) - a list of all the players that are currently watching the node
+
+Each blackboard also contains the following global data:
+
+resources (dict) - a dictionary with the names of resources as keys and the
+                   things using them as values. See the resource allocator
+                   node for details about this.
+
+errors (dict) - a dictionary with the hashes of nodes as keys and the error
+                strings they returned during the most recent iteration of the
+                blackboard as values.
+
+### Setting up the ai system
+
+First, subclass the classes of any AI-using objects, characters or other
+entities from the AIObject class:
+
+```
+from evennia.contrib.aisystem import AIObject
+
+class MyClass(AIObject, <other superclasses>)
+```
+
+Also set up any script classes that you want to have an AI blackboard by
+subclassing them from the AIScript class, or simply use the AIScript class
+directly.
+
+```
+from evennia.contrib.aisystem import AIScript
+
+class MyScript(AIScript, <other superclasses>)
+```
+
+You can assign each class that inherits from AIObject or AIScript a property
+called `aitree`, whose value can be either a BehaviorTree or a string that
+references the name or database id of a given tree. For instance:
+
+```
+from evennia.contrib.aisystem import AIScript
+
+class MyScript(AIScript, <other superclasses>)
+    aitree = 'fighter tree'
+```
+
+Also set up the player class of any players that you want to have access to
+the AI system commands by subclassing it from AIPlayer.
+
+```
+from evennia.contrib.aisystem import AIPlayer
```

Comment:
Sounds less likely to put AI on a Player, maybe instead use a fictive "Mob" AIObject as the example of choice here? Also, I suppose you must assign `self.ainodes.update(new_nodes)` also on the other entity types?